sonar-kotlin-plugin/src/main/java/org/sonarsource/kotlin/checks/HardcodedIpCheck.kt

```diff
@@ -85,7 +87,13 @@ private fun isValidIPV6(ipv6: String, ipv4: String?): Boolean {
 
 private fun isIPV4Exception(ip: String) = ip.startsWith("127.")
         || "255.255.255.255" == ip || "0.0.0.0" == ip || ip.startsWith("2.5.")
+        || isReservedDocumentationIp(ip) || isLocalIPv4MappedIPv6(ip)
 
-private fun isIPV6Exception(ip: String) = IPV6_LOOPBACK.matches(ip) || IPV6_NON_ROUTABLE.matches(ip)
+private fun isIPV6Exception(ip: String) = IPV6_LOOPBACK.matches(ip) || IPV6_NON_ROUTABLE.matches(ip) || isReservedDocumentationIp(ip)
+        || isLocalIPv4MappedIPv6(ip)
 
 private fun getCompressionSeparatorCount(str: String) = str.split("::").size - 1
+
+private fun isReservedDocumentationIp(str: String) : Boolean = RESERVED_DOC_IP_RANGE.any { it -> str.startsWith(it) }
+
+private fun isLocalIPv4MappedIPv6(str: String) : Boolean = LOCAL_IPV4_MAPPED_IPV6.any { it -> str.startsWith(it, true) }
```

Comment:
If you use `it` as parameter in anonymous functions such as `any`, you don't need to declare the parameter. `it` is the default name and you only need to declare the parameter explicitly, if you want to use a different name or do something more complex with the argument (e.g. destructure it). So for line 97, it would simply become
```
.any { str.startsWith(it) }
```
To make it more readable, you could use a more impactful name for `str`, such as `ip`. (I realize that the existing function on line 95 is not a good example here :))